app/controllers/users/totp_setup_controller.rb

```diff
@@ -0,0 +1,43 @@
+module Users
+  class TotpSetupController < ApplicationController
+    before_action :confirm_two_factor_authenticated
+
+    # GET /users/totp
+    def index
+      if session[:otp_secret_key].nil?
+        session[:otp_secret_key] = current_user.generate_totp_secret
+      end
+      @qrcode = UserDecorator.new(current_user).qrcode(session[:otp_secret_key])
+    end
+
+    # PATCH /users/totp
```

Comment:
@monfresh what do you think of this convention of adding the information as comments on the actions?  Should we always be doing this?